operators/src/source/ogr_source.rs

```diff
@@ -361,6 +364,50 @@ where
         }
     }
 
+    fn open_gdal_dataset(dataset_info: &OgrSourceDataset) -> Result<Dataset> {
+        // TODO: reliably detect CSV files or allow defining them as such in params
+        if let Some(ext) = dataset_info.file_name.extension() {
+            if ext == "csv" || ext == "tsv" {
+                let columns = dataset_info
+                    .columns
```

Comment:
Maybe we can split that into two functions to make it better readable and then do something like this:
```rust
match dataset_info.file_name.extension().and_then(OsStr::to_str) {
    Some("csv") | Some("tsv") => open_csv_dataset(dataset_info),
    _ => Ok(Dataset::open(&dataset_info.file_name)?),
}
```

Reply:
992dbbe0c995dc1ba9aef4947614ba1dbd93eb96